bus-mapping/src/evm/opcodes.rs

```diff
@@ -817,6 +821,56 @@ pub fn gen_end_tx_ops(state: &mut CircuitInputStateRef) -> Result<ExecStep, Erro
     Ok(exec_step)
 }
 
+// Add 3 RW read operations for transaction L1 fee.
+fn gen_tx_l1_fee_ops(state: &mut CircuitInputStateRef, exec_step: &mut ExecStep) {
+    let tx_id = state.tx_ctx.id();
+
+    let base_fee = Word::from(state.tx.l1_fee.base_fee);
```

Comment:
I think the values here should be read from `state.sdb` and **updated** into `state.tx` (since the tx is mutable here). There would be no fields related to l1fee inside eth trace. So if you do not update the `Transaction` struct all the value would be 0

Reply:
It seems that in [handle_tx](https://github.com/scroll-tech/zkevm-circuits/blob/develop/bus-mapping/src/circuit_input_builder.rs#L346) [gen_begin_tx_ops](https://github.com/scroll-tech/zkevm-circuits/blob/develop/bus-mapping/src/circuit_input_builder.rs#LL370C9-L370C25) is called after [self.new_tx](https://github.com/scroll-tech/zkevm-circuits/blob/develop/bus-mapping/src/circuit_input_builder.rs#L203).

The `l1_fee` and `l1_fee_committed` are fetched from state DB in [Transaction::new](https://github.com/scroll-tech/zkevm-circuits/pull/499/files#diff-78b4d906d4badeb0150c1619eda966b91019d143be0bd4338aa07074fc117686R331-R339) which is called by `new_tx`.